hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -245,7 +249,14 @@ public String extractPattern(TemporalUnit unit) {
 			case DAY_OF_YEAR:
 				return "({fn timestampdiff(sql_tsi_day, date(char(year(?2),4)||'-01-01'),?2)}+1)";
 			case DAY_OF_WEEK:
-				return "(mod({fn timestampdiff(sql_tsi_day, {d '2000-01-01'}, ?2)},7)+1)";
+				// Use the approach as outlined here: https://stackoverflow.com/questions/36357013/day-of-week-from-seconds-since-epoch
+				return "(mod(mod({fn timestampdiff(sql_tsi_day, {d '1970-01-01'}, ?2)}+4,7)+7,7)+1)";
+			case WEEK:
+				// Use the approach as outlined here: https://www.sqlservercentral.com/articles/a-simple-formula-to-calculate-the-iso-week-number
+				// In SQL Server terms this is (DATEPART(dy,DATEADD(dd,DATEDIFF(dd,'17530101',@SomeDate)/7*7,'17530104'))+6)/7
+				return "(({fn timestampdiff(sql_tsi_day, date(char(year(?2),4)||'-01-01'),{fn timestampadd(sql_tsi_day, {fn timestampdiff(sql_tsi_day, {d '1753-01-01'}, ?2)}/7*7, {d '1753-01-04'})})}+7)/7)";
```

Comment:
Since we had a test that needed the `WEEK` extraction, I looked into this topic and found the article I mentioned in the comment. The implementation/emulation works great, so I thought why not translate this for the Derby dialect.

Reply:
Good I looked into this again. Looks like I can now enable this test which failed before: https://github.com/hibernate/hibernate-orm/blob/2f1c7f60f7faea7d9947761cfc3b4ee97090bb05/hibernate-core/src/test/java/org/hibernate/orm/test/query/hql/FunctionTests.java#L824